src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -359,15 +370,45 @@ public void Train(double[] values, ref double[][] results)
                 {
                     Array.Resize<double[]>(ref results, values.Length);
                 }
-                SpectralResidual(values, results, _threshold);
+
+                bool isTemporal = true;
+                double[] seriesToDetect = values.ToArray();
+
+                if (_period > 0)
+                {
+                    StlConfiguration config = new StlConfiguration(_period);
+                    InnerStl stl = new InnerStl(values, config, isTemporal);
```

Comment:
>stl [](start = 29, length = 3)

Same question as above. Can this be a field in the class?

Reply:
updated.

---
In reply to: [435219659](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435219659) [](ancestors = 435219659)